core/src/main/java/com/graphhopper/routing/util/EncodedDoubleValue.java

```diff
@@ -31,6 +31,11 @@ public EncodedDoubleValue(String name, int shift, int bits, double factor, long
 
     public EncodedDoubleValue(String name, int shift, int bits, double factor, long defaultValue, int maxValue, boolean allowZero) {
         super(name, shift, bits, factor, defaultValue, maxValue, allowZero);
+        double factorDivision = maxValue / factor;
+        // Interestingly we cannot use modulo here, e.g. 10%00.1 != 0
+        if (factorDivision != (int) factorDivision) {
+            throw new IllegalStateException("MaxValue needs to be divideable by factor without remainder");
```

Comment:
> divideable -> divisible?

Yes, thank you.

> Maybe we can call checkValue somehow here to avoid separate code and include the other situations already in the constructor?

What do you mean by that?